### PR TITLE
feat: fee grant creation workflow

### PR DESCRIPTION
**Pull request by jlehtimaki**
This PR will add workflow to automatically create the fee grant when an operator is added to one of the `_IBC` paths

### CODE REVIEW

File: scripts/feegrant.sh [hunk 1]
```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env bash
+
+set -eu
+
+# Static variables
+DATE=$(date -d@"$((`date +%s`+2592000))" +"%Y-%m-%dT08:10:00Z")
+CHAIN_ID="archway-1"
+NODE_URL="https://rpc.mainnet.archway.io:443"
+ALLOWED_MESSAGES="/ibc.core.client.v1.MsgCreateClient,/ibc.core.client.v1.MsgUpdateClient,/ibc.core.client.v1.MsgUpgradeClient,/ibc.core.client.v1.MsgSubmitMisbehaviour,/ibc.core.client.v1.MsgRecoverClient,/ibc.core.client.v1.MsgIBCSoftwareUpgrade,/ibc.core.client.v1.MsgUpdateClientParams,/ibc.core.connection.v1.MsgConnectionOpenInit,/ibc.core.connection.v1.MsgConnectionOpenTry,/ibc.core.connection.v1.MsgConnectionOpenAck,/ibc.core.connection.v1.MsgConnectionOpenConfirm,/ibc.core.connection.v1.MsgUpdateConnectionParams,/ibc.core.channel.v1.MsgChannelOpenInit,/ibc.core.channel.v1.MsgChannelOpenTry,/ibc.core.channel.v1.MsgChannelOpenAck,/ibc.core.channel.v1.MsgChannelOpenConfirm,/ibc.core.channel.v1.MsgChannelCloseInit,/ibc.core.channel.v1.MsgChannelCloseConfirm,/ibc.core.channel.v1.MsgRecvPacket,/ibc.core.channel.v1.MsgTimeout,/ibc.core.channel.v1.MsgTimeoutOnClose,/ibc.core.channel.v1.MsgAcknowledgement,/ibc.applications.transfer.v1.MsgTransfer,/ibc.applications.transfer.v1.MsgUpdateParams"
```
Comment (kayano) [inline scripts/feegrant.sh:9]: do we need to allow all those messages?

File: scripts/feegrant.sh [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    exit 1
+  fi
+  if [[ "$GRANT" -le 0 ]]; then
+    echo "Did not find fee grant for ${operator}" >&2
+    create_grant
+  fi
+done
+
+echo "Done" >&2
+exit 0
```
Comment (kayano) [inline scripts/feegrant.sh:56]: not needed

File: scripts/feegrant.sh [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+# Checks every operator address if it can find active feegrant
+# if $GRANT length less or equal to 0, create the fee grant
+for operator in $OPERATORS; do
+  echo "Checking operator ${operator}" >&2
+  GRANT=$(archwayd q feegrant grants-by-grantee "$operator" --node https://rpc.mainnet.archway.io:443 -o json |jq '.allowances | length' )
+  if [[ -z "$GRANT" ]]; then
+    exit 1
+  fi
+  if [[ "$GRANT" -le 0 ]]; then
```
Comment (kayano) [inline scripts/feegrant.sh:49]:
```suggestion
  if [[ "$GRANT" -eq 0 ]]; then
```

File: scripts/feegrant.sh [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+# Authenticate
+echo "$MNEMONIC" | archwayd keys add mainnet-relayer-archway --keyring-backend test --recover
+
+# Start the check
+# Get all operators and if it's null exit with error
+OPERATORS=$(get_operators)
+if [[ "$OPERATORS" == "null" ]]; then
+  echo "Error: Could not find any operators" >&2
+  exit 1
```
[the remark this answers is not in the document]
Comment (jlehtimaki) [reply scripts/feegrant.sh:38]: Ahh, was under the impression that the operators need to be added.
Will change.

File: scripts/feegrant.sh (hunk 1 above)
Comment (jlehtimaki) [reply scripts/feegrant.sh:9]: They are based on the requirements in the ticket and taken from IBC Go proto:
https://github.com/cosmos/ibc-go/blob/a0ac240a2dc395eb65d1d3a312b592c61938c0be/proto/ibc/core/connection/v1/tx.proto#L14-L32

https://github.com/cosmos/ibc-go/blob/a0ac240a2dc395eb65d1d3a312b592c61938c0be/proto/ibc/core/client/v1/tx.proto#L14-L37

https://github.com/cosmos/ibc-go/blob/a0ac240a2dc395eb65d1d3a312b592c61938c0be/proto/ibc/core/channel/v1/tx.proto#L13-L45

**Review by kayano — APPROVED** [review]

**Pull request merged**